unliftio-core/src/Control/Monad/IO/Unlift.hs

```diff
@@ -131,12 +131,13 @@ underlying transformer.
 
 > newtype AppT m a = AppT { unAppT :: ReaderT Int (ResourceT m) a }
 >   deriving (Functor, Applicative, Monad, MonadIO)
->   -- Unfortunately, deriving MonadUnliftIO does not work.
 >
+> -- Same as `deriving newtype (MonadUnliftIO)`
 > instance MonadUnliftIO m => MonadUnliftIO (AppT m) where
 >   withRunInIO = wrappedWithRunInIO AppT unAppT
 -}
 {-# INLINE wrappedWithRunInIO #-}
+{-# DEPRECATED wrappedWithRunInIO "You can derive MonadUnliftIO for newtypes in unliftio-core 0.2.0.0 and later" #-}
```

Comment:
I think I'd rather leave off the deprecation warning here, it doesn't hurt to keep this function and warnings like this create a lot of headaches downstream.

Reply:
Sounds good 👍 Do you mind if I move the deprecation text into the docs? And if so, does the wording sound good to you?

Reply:
Sounds perfect